### ISSUE TITLE
Plugin version 0.21 and above requires Node 14

### ISSUE THREAD

**Issue by Cerber-Ursi** [SonarSource/eslint-plugin-sonarjs#433]
**I want to provide the following feedback.**

We were updating the relatively old project, which is for a moment stuck on (non-supported) Node.JS 12. According to the documentation, we were expecting the current version of plugin to still run on this version, but it actually fails with a syntax error - namely, it fails to parse the optional chaining operator in [`utils/jsx.js`](https://github.com/SonarSource/eslint-plugin-sonarjs/blob/master/src/utils/jsx.ts). If I understand correctly, this change was introduced [here](https://github.com/SonarSource/eslint-plugin-sonarjs/commit/88a748d8cec7ec316a05e9f51b5118136d43213c), via the bump of TypeScript target version.

I'm not sure whether it's the documentation or the output format that's incorrect, but right now they are disagreeing with each other, so one of them probably should be changed.

**Comment by yassin-kammoun-sonarsource** [comment]
Hello @Cerber-Ursi,

Thank you very much for bringing this to our attention!

Upon reviewing your feedback, we realized that there has been an oversight on our part in updating the documentation to reflect that the plugin now requires Node.js 14 or higher. This requirement is correctly stated in the [package.json](https://github.com/SonarSource/eslint-plugin-sonarjs/blob/649f6092cf2ba129b34feccaf860944df3a7734a/package.json#L19) file. Additionally, we have stopped testing the plugin against Node.js 12 on our Continuous Integration platform, as evidenced [here](https://github.com/SonarSource/eslint-plugin-sonarjs/pull/370).

To rectify this, I will promptly create a Pull Request to update the documentation accordingly. In the meantime, we kindly suggest that you either upgrade the Node.js runtime used by your project or consider using an older version of the plugin.

Cheers.
